### PR TITLE
better write lock timeout error

### PR DESCRIPTION
**Pull request by RasPhilCo**
Closes https://github.com/heroku/cli/issues/548

### CODE REVIEW

**Comment by RasPhilCo** [comment]
@ransombriggs @dickeyxxx

File: src/lock.js [hunk 1]
```diff
@@ -46,8 +46,16 @@ export default class Lock {
...
     // grab writer lock
-    let unlock = await lock.write(this.updatelockfile)
-    debug('upgraded to writer lock')
+    let unlock
+    try {
+      unlock = await lock.write(this.updatelockfile)
+      debug('upgraded to writer lock')
+    } catch (err) {
+      if (err.message.match(/is locked with(.)+reader/)) {
+        throw new Error('Command timed out waiting for other active process to finish')
```
Comment (jdx) [inline src/lock.js:55]: would be nice if it showed the pid

**Comment by jdx** [comment]
I think a better solution would also resolve https://github.com/heroku/cli-engine/issues/258. Which would involve a change to rwlockfile. I think it would be good if it showed a list of all the pids and what they were doing.

**Comment by RasPhilCo** [comment]
Lol, looking at the commits. Address this another time.

**Pull request closed without merging**